tfrecorder/schema.py

```diff
@@ -0,0 +1,179 @@
+    allowed_values=None)
+
+allowed_split_values = ['TRAIN', 'VALIDATION', 'TEST', 'DISCARD']
+split_key = _supported_type(
+    type_name='split_key',
+    feature_spec=tf.io.FixedLenFeature([], tf.string),
+    allowed_values=allowed_split_values)
+
+#TODO(mikebernico): Implement in preprocess_fn
+integer_input = _supported_type(
+    type_name='integer_input',
+    feature_spec=tf.io.FixedLenFeature([], tf.int64),
+    allowed_values=None)
+
+#TODO(mikebernico): Implement in preprocess_fn
+float_input = _supported_type(
+    type_name='float_input',
+    feature_spec=tf.io.FixedLenFeature([], tf.float64),
+    allowed_values=None)
+
+#TODO(mikebernico): Implement in preprocess_fn
+categorical_input = _supported_type(
+    type_name='categorical_input',
+    feature_spec=tf.io.FixedLenFeature([], tf.string),
+    allowed_values=None)
+
+integer_label = _supported_type(
+    type_name='integer_label',
+    feature_spec=tf.io.FixedLenFeature([], tf.int64),
+    allowed_values=None)
+
+string_label = _supported_type(
+    type_name='string_label',
+    feature_spec=tf.io.FixedLenFeature([], tf.string),
+    allowed_values=None)
+
+image_support_type = _supported_type(
+    type_name='image_support_type',
+    feature_spec=tf.io.FixedLenFeature([], tf.string),
+    allowed_values=None)
+
+# Default schema supports the legacy image_csv format.
+image_csv_schema = frozendict.FrozenOrderedDict({
+    'split': split_key,
+    'image_uri': image_uri,
+    'label': string_label})
+
+
+def get_raw_schema_map(
+    schema_map: Dict[str, collections.namedtuple]
+    ) -> Dict[str, collections.namedtuple]:
+  """Converts a schema to a raw (pre TFT / post image extraction) schema."""
+  raw_schema = {}
+  for k, v in schema_map.items():
+    if v.type_name == "image_uri":
+      raw_schema['image_name'] = image_support_type
+      raw_schema['image'] = image_support_type
+      raw_schema['image_height'] = image_support_type
+      raw_schema['image_width'] = image_support_type
+      raw_schema['image_channels'] = image_support_type
+    else:
+      raw_schema[k] = schema_map[k]
+  return raw_schema
+
+
+def get_tft_coder(columns: List[str],
+                  schema_map: Dict[str, collections.namedtuple]
+                  ) -> tft.coders.CsvCoder:
+  """Gets a TFT CSV Coder.
+
+  Args:
+    columns: Ordered dataframe column names, from df.column.
+    schema_map: Schema map used to infer the schema.
+
+  Returns:
+    tft.coders.CsvCoder
+  """
+  feature_spec = dict()
+
+  # Because the DF column name order may not match the feature_spec order
+  # This maps existing column names to their feature spec (req part of
+  # namedtuple)
+  for col in columns:
+    feature_spec[col] = schema_map[col].feature_spec
+
+
+  metadata = dataset_metadata.DatasetMetadata(
+      schema_utils.schema_from_feature_spec(feature_spec))
+  return tft.coders.CsvCoder(columns,
+                             metadata.schema)
+
+
+def get_key(
```

Comment:
nit: maybe this signature can fit in 2 lines.

Reply:
Ack